### PR TITLE
Add a logging screen test

### PR DESCRIPTION
**Pull request by DaveShuckerow**
Also, give more control to Table clients over how they create keys.  We don't need the PageStorageKey in most cases.

### CODE REVIEW

File: packages/devtools_app/test/flutter/logging_screen_test.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    testWidgets('can clear logs', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+      verifyNever(mockLoggingController.clear());
+      await tester.tap(find.text('Clear logs'));
+      verify(mockLoggingController.clear()).called(1);
+    });
+
+    testWidgets('can toggle structured errors', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+    });
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/flutter/logging_screen_test.dart]: add logic to toggle structured errors on/off

File: packages/devtools_app/test/flutter/logging_screen_test.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      await tester.pumpWidget(wrap(Builder(builder: screen.build)));
+      await tester.pumpAndSettle();
+      expect(find.byType(LogsTable), findsOneWidget);
+      expect(find.byKey(ValueKey(fakeLogData.last)), findsOneWidget,
+          reason: 'the most recently added logdata should show in the list by '
+              'default.');
+      expect(find.byKey(ValueKey(fakeLogData.first)), findsNothing,
+          reason:
+              'the least recently added logdata should be at the top of the '
+              'list, hidden from view.');
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/flutter/logging_screen_test.dart]: nit: trailing comma - here and in expect statement above

File: packages/devtools_app/test/flutter/logging_screen_test.dart (hunk 1 above)
Comment (DaveShuckerow) [reply packages/devtools_app/test/flutter/logging_screen_test.dart]: Good catch; done.

File: packages/devtools_app/test/flutter/logging_screen_test.dart (hunk 2 above)
Comment (DaveShuckerow) [reply packages/devtools_app/test/flutter/logging_screen_test.dart]: Done.

**Pull request merged**